plugins/in_nginx_exporter_metrics/nginx.c

```diff
@@ -0,0 +1,358 @@
+client_error:
+    flb_upstream_conn_release(u_conn);
+conn_error:
+    if (ret == 0 && ctx->is_up == false) {
+        cmt_gauge_set(ctx->connection_up, ts, 1.0, 0, NULL);
+        ctx->is_up = true;
+    } else if (ret != 0 && ctx->is_up == true) {
+        cmt_gauge_set(ctx->connection_up, ts, 0.0, 0, NULL);
+        ctx->is_up = false;
+    }
+    ret = flb_input_metrics_append(ins, NULL, 0, ctx->cmt);
+    if (ret != 0) {
+        flb_plg_error(ins, "could not append metrics");
+    }
+    return ret;
+}
+
+/**
+ * Function to initialize nginx_stub_status plugin.
+ *
+ * @param ins     Pointer to flb_input_instance
+ * @param config  Pointer to flb_config
+ *
+ * @return struct flb_in_nss_config* Pointer to the plugin's
+ *         structure on success, NULL on failure.
+ */
+struct nginx_ctx *nginx_ctx_init(struct flb_input_instance *ins,
+                                        struct flb_config *config)
+{
+    struct nginx_ctx *ctx;
+    struct flb_upstream *upstream;
+ 
+    if (ins->host.name == NULL) {
+        ins->host.name = flb_sds_create("localhost");
+    }
+    if (ins->host.port == 0) {
+        ins->host.port = 80;
+    }
+
+    ctx = flb_calloc(1, sizeof(struct nginx_ctx));
+    if (!ctx) {
+        flb_errno();
+        return NULL;
+    }
+    ctx->is_up = true;
+    
+    ctx->ins = ins;
+
+    ctx->cmt = cmt_create();
+    if (!ctx->cmt) {
+        flb_plg_error(ins, "could not initialize CMetrics");
+        flb_free(ctx);
+        return NULL;
+    }
+    
+    upstream = flb_upstream_create(config, ins->host.name, ins->host.port,
+                                 FLB_IO_TCP, NULL);
+    if (!upstream) {
+        flb_error("[nginx_stub_status] upstream initialization error");
+        return NULL;
+    }
+    ctx->upstream = upstream;
+
+    return ctx;
+}
+
+/**
+ * Callback function to initialize docker events plugin
+ *
+ * @param ins     Pointer to flb_input_instance
+ * @param config  Pointer to flb_config
+ * @param data    Unused
+ *
+ * @return int 0 on success, -1 on failure
+ */
+static int nginx_init(struct flb_input_instance *ins,
+                      struct flb_config *config, void *data)
+{
+    struct nginx_ctx *ctx = NULL;
+    
+    /* Allocate space for the configuration */
+    ctx = nginx_ctx_init(ins, config);
+    if (!ctx) {
+        return -1;
+    }
+
+    // https://github.com/nginxinc/nginx-prometheus-exporter#metrics-for-nginx-oss
+    ctx->connections_accepted = cmt_counter_create(ctx->cmt, "nginx", "connections", 
+        "accepted", "Accepted client connections", 0, NULL);
+    if (ctx->connections_accepted == NULL) {
+        return -1;
+    }
+    
+    ctx->connections_handled = cmt_counter_create(ctx->cmt, "nginx", "connections", 
+        "handled", "Handled client connections", 0, NULL);
+    if (ctx->connections_handled == NULL) {
+        return -1;
+    }
+    
+    ctx->connections_total = cmt_counter_create(ctx->cmt, "nginx", "http_requests", 
```

Comment:
I think here the proper namespace is `nginx`, subsystem is `http` and metric name `requests_total`.